cmd/kubeapps-apis/server/packages_test.go

```diff
@@ -0,0 +1,580 @@
+				Context: &corev1.Context{
+					Cluster:   "",
+					Namespace: globalPackagingNamespace,
+				},
+				PaginationOptions: &corev1.PaginationOptions{PageToken: "4", PageSize: 1},
+			},
+
+			expectedResponse: &corev1.GetAvailablePackageSummariesResponse{
+				AvailablePackageSummaries: []*corev1.AvailablePackageSummary{},
+				Categories:                []string{"cat-1"},
+				NextPageToken:             "",
+			},
+			statusCode: codes.OK,
+		},
+		{
+			name: "it should fail when calling the core GetAvailablePackageSummaries operation with one of the plugin failing",
+			configuredPlugins: []*pkgsPluginWithServer{
+				mockedPackagingPlugin1,
+				mockedFailingPackagingPlugin,
+			},
+			request: &corev1.GetAvailablePackageSummariesRequest{
+				Context: &corev1.Context{
+					Cluster:   "",
+					Namespace: globalPackagingNamespace,
+				},
+			},
+
+			expectedResponse: &corev1.GetAvailablePackageSummariesResponse{
+				AvailablePackageSummaries: []*corev1.AvailablePackageSummary{},
+				Categories:                []string{""},
+			},
+			statusCode: codes.NotFound,
+		},
+	}
+
+	for _, tc := range testCases {
+		t.Run(tc.name, func(t *testing.T) {
+			server := &packagesServer{
+				plugins: tc.configuredPlugins,
+			}
+			availablePackageSummaries, err := server.GetAvailablePackageSummaries(context.Background(), tc.request)
+
+			if got, want := status.Code(err), tc.statusCode; got != want {
+				t.Fatalf("got: %+v, want: %+v, err: %+v", got, want, err)
+			}
+
+			if tc.statusCode == codes.OK {
+				opt1 := ignoreUnexportedOpts
+				if got, want := availablePackageSummaries, tc.expectedResponse; !cmp.Equal(got, want, opt1) {
+					t.Errorf("mismatch (-want +got):\n%s", cmp.Diff(want, got, opt1))
+				}
+			}
+		})
+	}
+}
+
+func TestGetAvailablePackageDetail(t *testing.T) {
+	testCases := []struct {
+		name              string
+		configuredPlugins []*pkgsPluginWithServer
+		statusCode        codes.Code
+		request           *corev1.GetAvailablePackageDetailRequest
+		expectedResponse  *corev1.GetAvailablePackageDetailResponse
+	}{
+		{
+			name: "it should successfully call the core GetAvailablePackageDetail operation",
+			configuredPlugins: []*pkgsPluginWithServer{
+				mockedPackagingPlugin1,
+				mockedPackagingPlugin2,
+			},
+			request: &corev1.GetAvailablePackageDetailRequest{
+				AvailablePackageRef: &corev1.AvailablePackageReference{
+					Context: &corev1.Context{
+						Cluster:   "",
+						Namespace: globalPackagingNamespace,
+					},
+					Identifier: "pkg-1",
+					Plugin:     mockedPackagingPlugin1.plugin,
+				},
+				PkgVersion: "",
+			},
+
+			expectedResponse: &corev1.GetAvailablePackageDetailResponse{
+				AvailablePackageDetail: _test.MakeAvailablePackageDetail("pkg-1", mockedPackagingPlugin1.plugin),
+			},
+			statusCode: codes.OK,
+		},
+		{
+			name: "it should fail when calling the core GetAvailablePackageDetail operation with one of the plugin failing",
```

Comment:
Copy-n-paste error? I think this should be "it should fail when calling the core GetAvailablePackageDetail operation when the package is not present" or something?

Reply:
Renamed to _(...) when the package is not present in a plugin_. However, what I really wanted to highlight with this name is that, currently, the core APIs are failing if one (or more) plugin(s) fails. 

If, in the future, we add a fail-safe mode (eg. ignoring failing plugins or letting users configure the `strict: true/false` when enabling plugins), we might want to revisit these tests.